src/server/display_server.cpp

```diff
@@ -185,6 +191,16 @@ struct mir::DisplayServer::Private
         return true;
     }
 
+    void lock()
+    {
+        session_locker->request_lock();
+    }
+
+    void unlock()
+    {
+        session_locker->request_unlock();
+    }
+
```

Comment:
I don't think we need involve the `DisplayServer::Private` in forwarding these calls: `the_session_locker()` should be a dependency resolved by `the_console_services()`. (I.e. have it implement an interface used by `LogindConsoleServices` and supply it in `the_console_services()`.